Event list directive

slusarz opened the pull request:
Think this has reached a point of beta, so we can begin to iterate through design decisions, fact-checking, etc.

sirainen commented:
This fails with me locally:

```
Exception occurred:
  File "/home/parallels/cvs/documentation-tss/source/_ext/dovecot_sphinx.py", line 213, in add_entry
    self.data['entry'][signature] = (self.env.docname, anchor, self.label)
KeyError: 'entry'

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/usr/local/lib/python3.8/dist-packages/sphinx/application.py", line 343, in build
    self.builder.build_update()
  File "/usr/local/lib/python3.8/dist-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/usr/local/lib/python3.8/dist-packages/sphinx/builders/__init__.py", line 307, in build
    updated_docnames = set(self.read())
  File "/usr/local/lib/python3.8/dist-packages/sphinx/builders/__init__.py", line 414, in read
    self._read_serial(docnames)
  File "/usr/local/lib/python3.8/dist-packages/sphinx/builders/__init__.py", line 435, in _read_serial
    self.read_doc(docname)
  File "/usr/local/lib/python3.8/dist-packages/sphinx/builders/__init__.py", line 475, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/usr/local/lib/python3.8/dist-packages/sphinx/io.py", line 189, in read_doc
    pub.publish()
  File "/usr/local/lib/python3.8/dist-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/usr/local/lib/python3.8/dist-packages/sphinx/io.py", line 109, in read
    self.parse()
  File "/usr/local/lib/python3.8/dist-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/usr/local/lib/python3.8/dist-packages/sphinx/parsers.py", line 101, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 3008, in text
    self.section(title.lstrip(), source, style, lineno + 1, messages)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 3008, in text
    self.section(title.lstrip(), source, style, lineno + 1, messages)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/usr/local/lib/python3.8/dist-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/usr/local/lib/python3.8/dist-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/usr/local/lib/python3.8/dist-packages/sphinx/domains/__init__.py", line 286, in run
    return super().run()
  File "/usr/local/lib/python3.8/dist-packages/sphinx/directives/__init__.py", line 200, in run
    self.add_target_and_index(name, sig, signode)
  File "/home/parallels/cvs/documentation-tss/source/_ext/dovecot_sphinx.py", line 349, in add_target_and_index
    self.env.get_domain(self.domain).add_entry(sig, anchor)
  File "/home/parallels/cvs/documentation-tss/source/_ext/dovecot_sphinx.py", line 213, in add_entry
    self.data['entry'][signature] = (self.env.docname, anchor, self.label)
KeyError: 'entry'
```

slusarz commented:
I've had issues before with caching and a first run of make failing after python code has changed.  Running make again and/or clearing cache (i.e. make clean) rebuilds all files and always worked for me.

I'm building on Python 3.8.10 on local machine.  Just did a reinstall of requirements.txt, and compilation works fine.

sirainen commented:
Yeah, looks like make clean helped. About the output .. I'm not sure I like the collapse/expand behavior. It makes searching text more difficult. Maybe could be ok if there were buttons to collapse/expand everything at once.

slusarz commented:
> Yeah, looks like make clean helped. About the output .. I'm not sure I like the collapse/expand behavior. It makes searching text more difficult. Maybe could be ok if there were buttons to collapse/expand everything at once.

I've updated this MR to match recent main branch changes (... we should really try to resolve this somewhat soon, since these merges are complicated and annoying ...).

Also, I've added buttons to allow for Expand and Collapse all behavior.  Could probably extend a bit to save the choice with cookies or something, but that's starting to get to be a bunch of javascript code - since the JS libraries already included in the page (e.g. jquery) don't support cookies by default, so we would have to include additional js files.  So I kept it as simple as possible for this initial review.

sirainen commented:
Maybe we could already merge, but ..:
 * It adds a bunch of TODOs in place of where previously there was just no description for event.
 * Also a bunch of .. todo items.
 * The section titles don't look like ones, because event names font is almost as large and they are in bold. So it looks a bit confusing.

slusarz commented:
> * It adds a bunch of TODOs in place of where previously there was just no description for event.

This is intentional :)  If we are going to rewrite the page, we should ensure that content is there.

Agreed, though, that this is blocking on merging this.  We will need to give out homework to have someone at least come up with some kind of placeholder.

> * Also a bunch of .. todo items.

See above.  I think we need to at least do a pass to see if we can clear up those confusions.

As someone who is reasonably knowledgable on events, elements with the todo directives were so confusing to me that I became physically frustrated.  So that is unacceptable for a target group that will likely have even less knowledge.  Especially for events, if we are not going to document them in a way people can use it, then we should just take those events out.

> * The section titles don't look like ones, because event names font is almost as large and they are in bold. So it looks a bit confusing.

I think this is more of the (ongoing) battle with sphinx/our theme not allowing or implementing the best design.  Semantically, we are using exactly what is being asked by the software, but I agree that the presentation is not great.  I guess we can dig into custom CSS to band-aid this, although this is more of an annoyance than a show-stopper.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
 
-*no particular fields*
+.. dovecot_core:event:: auth_client_cache_flush_started
+   :added: v2.3.7
```
I think we should just remove this. I don't see any use for this. Same with _finished. This event is currently sent by "doveadm auth cache flush" command. It could make sense to have "auth_cache_flush_finished" event on the auth server side, but it makes no sense to have a client-side event.

slusarz replied:
DOP-2869, we should remove in 3.0.  I'll mark as deprecated/removed in this MR and just put placeholder text for the TODO.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+   :added: v2.3.7
+   :inherit: auth_client_lookup
+
+   TODO
```
The name is rather self-explanatory. Auth client does a passdb lookup.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+
+   :field user_mask: User mask to list.
+
+   TODO
```
Auth client starts userdb iteration.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+   :added: v2.3.7
+   :inherit: auth_client_lookup
+
+   TODO
```
Auth client does userdb lookup.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+   :added: v2.3.7
+   :inherit: auth_client_request
+
+   TODO
```
Auth client starts authentication request.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+   :added: v2.3.7
+   :inherit: auth_client_request
+
+   TODO
```
Auth client receives request from server to continue SASL authentication.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+   :added: v2.3.7
+   :inherit: auth_client_request
+
+   TODO
```
Auth client continues SASL authentication by sending a response to server request.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,284 @@ SQL Categories
+   :added: v2.3.7
+   :inherit: auth_client_request
+
+   TODO
```
Auth client receives response from server that authentication is finished, either success or failure.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +482,384 @@ These events apply only for connections using the ``connection API``.
-+=====================+======================================================+
-| Inherits from environment (e.g. :ref:`event_mail_user`)                    |
-+---------------------+------------------------------------------------------+
+   .. todo:: WHAT DOES THIS MEAN???
```
Means it could be inherited from all kinds of different things. I guess it could be said differently, like `May be inherited from various different parents (e.g. :mail_user event) or even from no parent.`

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +482,384 @@ These events apply only for connections using the ``connection API``.
+   Inherits from fs or any other specified event (e.g. mail).
 
-.. _event_fs_iter:
+   .. todo:: WHAT DOES THIS MEAN???
```
Pretty much what it says. The parent event comes from a parameter, so it can be anything.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +482,384 @@ These events apply only for connections using the ``connection API``.
+
+   If the file was created for obox, it also has fields:
+
+   .. todo:: WHAT DOES THIS MEAN???
```
I guess you meant it for the "Inherits .." text, same explanation here.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +482,384 @@ These events apply only for connections using the ``connection API``.
+
+   Inherits from fs or any other specified event (e.g. mailbox).
+
+   If the file was created for obox, it also has fields:
```
Hmm. Something's wrong here now. The "file_type" and "reason" are supposed to be explained below this, they don't exist for all fs_files. And this is supposed to be for fs_file event, not for fs_iter event. Looks like that was wrong already in the original page.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1435,354 +868,222 @@ HTTP Server
 
-dns_request_started
--------------------
+   TODO
```
DNS request is started to be processed by the dns worker process.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1435,354 +868,222 @@ HTTP Server
 
-dns_worker_request_finished
----------------------------
+   TODO
```
DNS request is sent by the DNS client library to the DNS worker process.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1792,581 +1093,421 @@ Events emitted by Dovecot's internal SQL library.
 
-sql_query_finished
-------------------
+   TODO
```
Response was received to SQL query.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1792,581 +1093,421 @@ Events emitted by Dovecot's internal SQL library.
-+---------------------+------------------------------------------------------+
-| error_code          | Error code (if available)                            |
-+---------------------+------------------------------------------------------+
+   TODO
```
SQL transaction was committed or rolled back.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1792,581 +1093,421 @@ Events emitted by Dovecot's internal SQL library.
------------------------
+.. dovecot_core:event:: sql_connection_finished
+
+   TODO
```
Connection to SQL server is closed. I don't know if this event is of any use.

slusarz replied:
I think it theoretically might be useful to debug/monitor how often Dovecot is connecting to the external service.  Regardless, this seems small and straightforward enough that I don't see much utility in removing it.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -2378,561 +1519,525 @@ obox plugin
+
+   Changes in metacache are being uploaded to storage.
+
+   .. todo:: Does this inherit from mail_user or mailbox?
```
Either. Depends on whether metacache root bundle is uploaded or mailbox bundle is being up loaded.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -2378,561 +1519,525 @@ obox plugin
+   :added: v2.3.11
+   :inherit: mail_user, metacache_clean
+
+   TODO
```
User is started to be cleaned from metacache, either entirely or only partially (only low priority files).

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -2378,561 +1519,525 @@ obox plugin
 
-dict_server_lookup_finished
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   TODO
```
dict library finished iterating keys?..

sirainen commented:
@slusarz ping

cmouse commented:
@slusarz ping

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,289 @@ SQL Categories
-=====================
+   :field passdb: Driver name.
+   :field passdb_name: ``passdb { name }``, if it is configured. Otherwise,
+     same as ``passdb``.
```
This sounds confusing here. The original text was "Otherwise, the driver name." meaning passdb { driver }. So maybe here also say "same as passdb { driver }"?

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,289 @@ SQL Categories
+
+   :field userdb: Driver name.
+   :field userdb_name: ``userdb { <name> }``, if it is configured. Otherwise,
+     same as ``userdb``.
```
same as userdb { driver }

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -164,646 +164,289 @@ SQL Categories
+.. dovecot_event:field_group:: auth_server_userdb
+
+   :field userdb: Driver name.
+   :field userdb_name: ``userdb { <name> }``, if it is configured. Otherwise,
```
\<name\> looks confusing here, since it's about the "name" setting inside userdb. Also passdb { name } was used above. So better remove brackets here.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
 
-fs
---
+   May be inherited from various different parents (e.g. "Mail User" event)
```
Why not link to "mail user" event?

slusarz replied:
There is no "mail user" event.  There's a inherited group of attribute/fields called "mail_user", but it can't be linked to separately.  And regardless, should we really be linking just mail user here?  If we were to add links, we would need to somehow list all the things that could be inherited from (or else people are just going to assume that mail user is the only thing that matter here).

And this leads to my biggest current issue with events - these polymorphic events are extremely confusing.  i.e. an event can have different fields based on what triggered it.  This is very troublesome for anybody who is trying to create input filters or parsers on a system that can't handle different fields for the same event.

sirainen replied:
>  There's a inherited group of attribute/fields called "mail_user", but it can't be linked to separately

Why can't we link to it?

>  And regardless, should we really be linking just mail user here? If we were to add links, we would need to somehow list all the things that could be inherited from (or else people are just going to assume that mail user is the only thing that matter here).

Could be a problem, yes..

> And this leads to my biggest current issue with events - these polymorphic events are extremely confusing. i.e. an event can have different fields based on what triggered it. This is very troublesome for anybody who is trying to create input filters or parsers on a system that can't handle different fields for the same event.

Well, that can't really be helped. We can't either create tons of different events depending on what the code path was to create them. I haven't heard of this being an issue though. Is it a real problem somewhere? I'd think that even in that case it should be possible to configure, just not configure `event=foo` but rather checking more fields like `event=foo AND NOT box=""`

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
 
-fs_file
--------
+   Inherits from fs or any other specified event (e.g. mail).
```
Why not link to mail event?

slusarz replied:
See above.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
+                     * ``fts``: FTS file
+   :field reason: Reason for accessing the file.
+
+   Inherits from fs or any other specified event (e.g. mailbox).
```
Why not link to mailbox?

slusarz replied:
See above.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
+                     * ``index``: Index bundle
+                     * ``box``: Mailbox directory (for creating/deleting it,
+                       if used by the storage driver)
+                     * ``fts``: FTS file
```
There is no indication that the file_type field exists only for files accessed via obox.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
+                     * ``box``: Mailbox directory (for creating/deleting it,
+                       if used by the storage driver)
+                     * ``fts``: FTS file
+   :field reason: Reason for accessing the file.
```
This reason is also obox-specific

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
+   Inherits from fs or any other specified event (e.g. mailbox).
+
+   .. todo:: obox events also have additional fields that need to be defined
+             here
```
I don't think there are additional fields, they were already above, but just not marked as such.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
-| session             | Session ID for the storage session                   |
-+---------------------+------------------------------------------------------+
+   .. todo:: mail_storage_service_user "Inherits from environment (e.g.
+             IMAP/LMTP client)"
```
Like done elsewhere, how about just write text:

```
Inherits from any specified event (e.g. imap client).
```

slusarz replied:
See above.  (I don't know what to do with these polymorphic events.)

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
-| uid                 | UID of the expunged mail.                              |
-+---------------------+--------------------------------------------------------+
+   A mail was expunged from the mailbox. Note that this event inherits from
+   mailbox, not mail.
```
mailbox & mail could be links?

slusarz replied:
See above: there is nothing to link to, as the inherited group is meant for internal use only (it's not a link on the page).

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
-| :ref:`event_mail_user` depending on what the index is used for.            |
-+---------------------+------------------------------------------------------+
+.. todo:: mail index "Inherits from event_mailbox, event_storage or
+          event_mail_user depending on what the index is used for."
```
Could be also a text with links to the common events.

slusarz replied:
See above.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
+   A mail index file was recreated.
 
-.. versionadded:: 2.3.12
+   .. todo:: list all possible reasons
```
The reason is human-readable string, can't list all possible reasons. Could modify to say it:

```
   :field reason: Human-readable reason why the mail index was recreated.
```

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
 
-Mail cache
-----------
+   .. todo:: Inherit from mail_index
```
I guess needs mail_index_common. It doesn't have any fields though.

slusarz replied:
Created an empty field_group named mail_index_common and had all events with this todo inherit from it.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
-^^^^^^^^^^^^^^^^^^^^^^
+.. dovecot_core:event:: mail_expunge_requested
+   :added: v2.3.15
+   :inherit: mail, mailbox
```
Does this need to list all inherited events? Couldn't it automatically go through the tree to list them all here by just specifying the first? Especially because now it stops at mailbox, why not continue to mail_user, mail_storage_user, ...

slusarz replied:
Given my comments elsewhere, I agree that this page could be adopted and extended to work much better with inheritance.  But someone else needs to do that who has more insight to me (since I can't understand inheritance from the old documentation.)

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1435,354 +869,222 @@ HTTP Server
 
-Common fields
--------------
+.. todo:: "Inherits from event_incoming_conn"
```
`:inherit: client_connection_common`?

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -1435,354 +869,222 @@ HTTP Server
 
-Common fields:
+.. dovecot_event:field_group:: mail_delivery
+   :inherit: smtp_recipient
```
not true if emitted by dovecot-lda..

slusarz replied:
See above.  (I don't see how this kind of event is useful unless the assumption is made that the event receiving client is able to accept different values for the same event.  In other words, a single event ID can have multiple payload signatures.  At a minimum, there needs to be a decision on how to document this - This would require some kind of dummy event link that doesn't currently exist...)

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -2378,561 +1521,524 @@ obox plugin
+   User index bundle file is being uploaded. Can happen while the user is
+   being uploaded.
+
+   .. note:: This event can be inherited from a mailbox event, and include the
```
link to mailbox event?

slusarz replied:
See above.

sirainen commented on source/admin_manual/list_of_events.rst:
```diff
@@ -2378,561 +1521,524 @@ obox plugin
+   User index bundle file was uploaded. Can happen while the user is being
+   uploaded.
+
+   .. note:: This event can be inherited from a mailbox event, and include the
```
link to mailbox event?

slusarz replied:
See above.

slusarz commented:
I have resolved all non-inheritance based comments.

But I worry that we are a long way away from committing this if we have to fully resolve inheritance issues right now.  Inheritance is confusing and not well documented on the current page so I don't think this is any worse.  Alternatively, we need to get together and work this out as a team, as I still don't fully understand inheritance patterns so it is difficult for me to "fix" these complaints.

sirainen commented:
> But I worry that we are a long way away from committing this if we have to fully resolve inheritance issues right now. Inheritance

I think we should just merge this ASAP and fix things later. I'm almost tempted to just merge it without reviewing :) But maybe I should do some minimal review at least..
